presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestPrestoNativeAsyncDataCacheCleanupAPI.java

```diff
@@ -0,0 +1,151 @@
+import static org.testng.Assert.assertNotEquals;
+
+public class TestPrestoNativeAsyncDataCacheCleanupAPI
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected QueryRunner createQueryRunner() throws Exception
+    {
+        return PrestoNativeQueryRunnerUtils.createQueryRunner(true);
+    }
+
+    @Override
+    protected ExpectedQueryRunner createExpectedQueryRunner() throws Exception
+    {
+        return PrestoNativeQueryRunnerUtils.createJavaQueryRunner();
+    }
+
+    @Override
+    protected void createTables()
+    {
+        QueryRunner queryRunner = (QueryRunner) getExpectedQueryRunner();
+        createCustomer(queryRunner);
+    }
+
+    @Test
+    public void testAsyncDataCacheCleanup() throws Exception
+    {
+        String metricsEndPoint = "/v1/info/metrics";
+        String cacheCleanupEndPoint = "/v1/memory";
+
+        QueryRunner queryRunner = getQueryRunner();
+        DistributedQueryRunner distributedQueryRunner = (DistributedQueryRunner) queryRunner;
+        Set<InternalNode> workerNodes = getWorkerNodes(distributedQueryRunner);
+
+        // Collect initial cache metrics
+        Metrics initialMetrics = collectCacheMetrics(workerNodes, metricsEndPoint);
+        assertEquals(0, initialMetrics.hits, "Cache hits should be zero initially.");
+        assertEquals(0, initialMetrics.entries, "Cache entries should be zero initially.");
+
+        // Execute queries to populate cache
+        for (int i = 0; i < 4; i++) {
+            assertQuery("SELECT count(*) FROM customer");
+        }
+        sleep(60000); // Sleep to allow cache updates
+
+        // Collect cache metrics after queries
+        Metrics populatedMetrics = collectCacheMetrics(workerNodes, metricsEndPoint);
+        assertNotEquals(0, populatedMetrics.hits, "Cache should have hits after queries.");
+        assertNotEquals(0, populatedMetrics.entries, "Cache should have entries after queries.");
+
+        // Trigger async cache cleanup
+        workerNodes.parallelStream().forEach(worker -> {
+            int responseCode = sendWorkerRequest(worker.getInternalUri().toString(), "\"CLEAN_ASYNC_DATA_CACHE\"", cacheCleanupEndPoint);
+            assertEquals(200, responseCode, "Expected a 200 OK response for cache cleanup.");
+        });
+        sleep(60000); // Sleep to allow cache updates
+
+        // Validate cache is cleared
+        Metrics finalMetrics = collectCacheMetrics(workerNodes, metricsEndPoint);
+        assertEquals(0, finalMetrics.entries, "Cache should be empty after cleanup.");
+    }
+
+    private Metrics collectCacheMetrics(Set<InternalNode> workerNodes, String endpoint)
+            throws Exception
+    {
+        int hits = 0;
+        int entries = 0;
+        for (InternalNode worker : workerNodes) {
+            Map<String, Long> metrics = fetchScalarLongMetrics(worker.getInternalUri().toString(), endpoint, "GET");
+            hits += metrics.getOrDefault("velox_memory_cache_num_hits", 0L);
```

Comment:
Where are the configs `velox_memory_cache_num_hits` and `velox_memory_cache_num_entries` defined?

Reply:
These counters should be defined on Velox side